### PR TITLE
Don't instantiate Faker on Seeder constructor

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
This attempts to hasten database live tests by extracting `Faker` instantiation from `Seeder`'s constructor to a new static method and using a static property so that it will persist across instances. I've added tests for uncovered lines.

My only worry for this is that the `use` statement for `Faker\Factory` may break if the library is not installed for manual installations. @samsonasik I've tried using `@noRector` but still it imports them. Any ideas?

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by samsonasik** [comment]
`use` statement won't break existing manual installation as far as there is a `class_exists()` check, it is just statement.

**Comment by samsonasik** [comment]
For `@noRector` usage, you can do like the following:

```php
/** @noRector \Utils\Rector\UnderscoreToCamelCaseVariableNameRector */
$a_b_c;
```

**Comment by paulbalandan** [comment]
Thanks for clarifying. So I'm just fearing for nothing. 🤣

File: system/Database/Seeder.php [hunk 1]
```diff
@@ -126,7 +126,21 @@ public function __construct(Database $config, BaseConnection $db = null)
...
 		$this->forge = Database::forge($this->DBGroup);
-		$this->faker = class_exists(Factory::class) ? Factory::create() : null;
+	}
+
+	/**
+	 * Gets the Faker Generator instance.
+	 *
+	 * @return \Faker\Generator|null
+	 */
+	public static function faker()
```
Comment (samsonasik) [inline system/Database/Seeder.php:136]: You change with this:

```suggestion
      public static function faker(): ?\Faker\Generator
```

and run :

```bash
vendor/bin/rector process system/Database/Seeder.php
```

so it should will automatically add use statement for `use Faker\Generator` to use statements. It doesn't sort use statements so probably need manual check or use tools like php-cs-fixer to sort it, eg:

```php
php-cs-fixer fix system/Database/Seeder.php --rules=ordered_imports
```

**Comment by michalsn** [comment]
Can we add info about `Faker` being available (and example how to access it) to the user guide? I think I missed it last time.

**Review by michalsn — APPROVED** [review]

**Pull request merged**